File: Domain/Entities/Product.cs
namespace Domain.Entities;

public sealed class Product
{
  private Product(){}

  public Product(string name, string description, decimal price, int stockQuantity)
  {
    Name = name;
    Description = description;
    Price = price;
    StockQuantity = stockQuantity;
    IsActive = true;
    CreatedAt = DateTime.UtcNow;
  }

  public int Id { get; private set; }

  public string Name { get; private set; } = string.Empty;

  public string Description { get; private set; } = string.Empty;

  public decimal Price { get; private set; }

  public int StockQuantity { get; private set; }

  public bool IsActive { get; private set; }

  public DateTime CreatedAt { get; private set; }

  public byte[] RowVersion { get; private set; } = [];
}

File: Frontend/src/types/product.ts
export type Product = {
  id: number;
  name: string;
  description: string | null;
  price: number;
  stockQuantity: number;
  isActive: boolean;
  createdAt: string;
};

export type CreateProductRequest = {
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
};

export type UpdateProductRequest = {
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
};

export type ProductListParams = {
  page?: number;
  pageSize?: number;
  name?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
};